File: programmers.com/learn/courses/30/lessons/43165.ts
export default function solution(numbers: number[], target: number) {
  const go = (index: number, cnt: number): number => {
    if (index == numbers.length) {
      if (cnt == target) return 1
      return 0
    }
    return (
      go(index + 1, cnt + numbers[index]) + go(index + 1, cnt - numbers[index])
    )
  }

  return go(0, 0)
}

File: programmers.com/learn/courses/30/lessons/43165.py
def solution(numbers: list, target: int):
    end = len(numbers)

    def go(index: int, acc: int):
        if index == end:
            if acc == target:
                return 1
            return 0
        return go(index+1, acc-numbers[index]) + go(index+1, acc+numbers[index])

    return go(0, 0)


def test_solution():
    assert solution([1, 1, 1, 1, 1], 3) == 5
